### PR TITLE
Osx build

### PR DESCRIPTION
**Pull request by twangboy**
Split the build script up into different tasks:
- `build_dev.sh` will build a dev environment with everything but salt
- `build_pkg.sh` will take the contents of `/opt/salt` and create a mac osx .pkg installer
- `build_sig.sh` will sign a package
- 'build.sh`runs`build_dev`, installs salt, and then runs`build_pkg`

Broke requirements file into a regular and an external requirements
- `req.txt` : packages that require no additional parameters
- `req_ext.txt` : packages that require additional parameters to install correctly

### CODE REVIEW

File: pkg/osx/req_ext.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+cryptography==1.1.2
+pyOpenSSL==0.15.1
+pygit2==0.23.2
+
```
Comment (thatch45) [inline pkg/osx/req_ext.txt]: What does req_ext mean? We can't distribute gpl libs (pygit2) easily, are these libs included in the installer?

**Comment by cachedout** [comment]
@twangboy Please rebase this so that all commits are under a valid GitHub username. After we get the concerns about the licensing sorted out and it's rebased, we should be able to merge this. Thanks.

File: pkg/osx/req_ext.txt (hunk 1 above)
Comment (twangboy) [inline pkg/osx/req_ext.txt]: These three require different parameters to install than the rest. They depend on openssl, libgit2, etc that require specific paths be passed as parameters.

**Comment by twangboy** [comment]
I'm gonna need a class on rebase... I was getting all kinds of conflicts I didn't know how to resolve. This has been resubmitted as #29904

**Pull request closed without merging**